azurerm/resource_arm_hd_insight_clusters.go

```diff
@@ -0,0 +1,918 @@
+package azurerm
+
+// Code based on the terraform.provider plugin by Microsoft (R) AutoRest Code Generator.
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/hdinsight/mgmt/2015-03-01-preview/hdinsight"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmHDInsightClusters() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmHDInsightClustersCreate,
+		Read:   resourceArmHDInsightClustersRead,
+		Update: resourceArmHDInsightClustersUpdate,
+		Delete: resourceArmHDInsightClustersDelete,
+		Schema: map[string]*schema.Schema{
+			"cluster_type": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"hadoop",
+					"hbase",
+					"storm",
+					"spark",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+			"cluster_version": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+			},
+			"component_version": {
+				Optional: true,
+				ForceNew: true,
+				Type:     schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"connectivity_endpoints": {
```

Comment:
can we move all of the computed fields to the end of the schema, we do it in the other resources to make it simpler to read